File: src/Lista1/Questao06.java
package Lista1;

import java.util.Scanner;

public class Questao06 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        sc.nextLine();

        String[] palavras = new String[n];

        for (int i = 0; i < n; i++) {
            palavras[i] = sc.nextLine();
        }

        String palavraProcurada = sc.nextLine();

        for (int i = 0; i < n; i++) {
            if (palavraProcurada.equals(palavras[i])) {
                System.out.println("palavra correta");
                return;
            } else if (palavraProcurada.length() == palavras[i].length() && (palavraProcurada.charAt(0) == palavras[i].charAt(0) || palavraProcurada.charAt(0) == palavras[i].charAt(1))) {
                for (int j = 0; j < palavraProcurada.length() - 1; j++) {
                    StringBuilder sb = new StringBuilder(palavraProcurada);
                    char c1 = palavraProcurada.charAt(j);
                    char c2 = palavraProcurada.charAt(j + 1);
                    sb.setCharAt(j, c2);
                    sb.setCharAt(j + 1, c1);

                    if (sb.toString().equals(palavras[i])) {
                        System.out.println("palavra incorreta");
                        System.out.println(palavras[i] + "?");
                        return;
                    }
                }
            }

        }
        System.out.println("palavra incorreta");
        System.out.println("nenhuma sugestão");
        sc.close();
    }
}

File: Listas C++/Lista 1/questao06.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
#include <locale>

int main() {
    
    std::cin.imbue(std::locale("C"));
    
    int n;
    std::cin >> n;
    std::cin.ignore();

    std::vector<std::string> palavras(n);

    for (int i = 0; i < n; i++) {
        std::getline(std::cin, palavras[i]);
    }

    std::string palavraProcurada;
    std::getline(std::cin, palavraProcurada);

    for (int i = 0; i < n; i++) {
        
        if (palavraProcurada == palavras[i]) {
            std::cout << "palavra correta" << std::endl;
            return 0;
        } 
        else if (palavraProcurada.length() == palavras[i].length() && (palavraProcurada.length() > 0 && (palavraProcurada[0] == palavras[i][0] || palavraProcurada[0] == palavras[i][1]))) {
            
            for (size_t j = 0; j < palavraProcurada.length() - 1; j++) {

                std::string tentativa = palavraProcurada;
                std::swap(tentativa[j], tentativa[j + 1]);

                if (tentativa == palavras[i]) {
                    std::cout << "palavra incorreta" << std::endl;
                    std::cout << palavras[i] << "?" << std::endl;
                    return 0;
                }
            }
        }
    }
    
    std::cout << "palavra incorreta" << std::endl;
    std::cout << "nenhuma sugestão" << std::endl;
    
    return 0;
}